src/replication.c

```diff
@@ -1380,14 +839,153 @@ handle_write_data_event(replica_t *replica)
 	return rc;
 }
 
+static void
```

Comment:
caller seems to be taking same lock and released before calling this function.
So, this can be merged without doing lock/unlock sequence twice.